pennylane_lightning/core/src/simulators/lightning_tensor/tncuda/TNCudaBase.hpp

```diff
@@ -265,8 +267,18 @@ class TNCudaBase : public TensornetBase<PrecisionT, Derived> {
             "Unsupported gate: MPS method only supports 1, 2-wires gates");
 
         auto &&par = (params.empty()) ? std::vector<PrecisionT>{0.0} : params;
-        DataBuffer<PrecisionT, int> dummy_device_data(
-            Pennylane::Util::exp2(wires.size()), getDevTag());
+
+        int64_t dummy_id = gate_ids_.empty() ? 1 : *gate_ids_.rbegin() + 1;
+
+        if (!gate_matrix.empty()) {
+            auto gate_key = std::make_pair(opName, par);
+            std::vector<CFP_t> matrix_cu =
+                cuUtil::complexToCu<ComplexT>(gate_matrix);
+            gate_cache_->add_gate(dummy_id, gate_key, matrix_cu, adjoint);
+        } else {
+            gate_cache_->add_gate(dummy_id, opName, par, adjoint);
+        }
```

Comment:
```suggestion
        if (gate_matrix.empty()) {
            gate_cache_->add_gate(dummy_id, opName, par, adjoint);
        } else {
            auto gate_key = std::make_pair(opName, par);
            std::vector<CFP_t> matrix_cu =
                cuUtil::complexToCu<ComplexT>(gate_matrix);
            gate_cache_->add_gate(dummy_id, gate_key, matrix_cu, adjoint);
        }
```
For good practices, always is better to work in a positive.

Reply:
A positive environment is good for the team's morale. [jk]